Tensile/Source/client/source/Reference.cpp

```diff
@@ -118,6 +118,34 @@ namespace Tensile
             throw std::runtime_error(msg.c_str());
         }
 
+        /* Transform the data type of A/B to Accumulator if A!=ACC or B!=ACC, but filter out cases, I8/I32/I32 and I8x4/I32/I32
+         * 
+         * There are three cases of doing multiplication and their conditions to do transform or not are as below.
+         * 1. AxB : (A!=ACC or B!=ACC) and A!=I8 and A!=I8x4
+         * 2. Alpha x rC :  (Alpha!=ACC or rC!=ACC)
+         * 3. Beta x C : (Beta!=ACC or C!=ACC)
+        */
+        template <typename A, typename B, typename Accumulator>
+        constexpr bool need_transform
+            = !(std::is_same<A, Accumulator>() && std::is_same<B, Accumulator>())
+              && !std::is_same<A, Int8>() //case I8/I32/I32
+              && !std::is_same<A, Int8x4>(); //case I8x4/I32/I32
+
+        template <typename A, typename B, typename Accumulator>
+        inline Accumulator multiply(
+            A a, B b, typename std::enable_if<!need_transform<A, B, Accumulator>>::type* = nullptr)
+        {
+            return static_cast<Accumulator>(a * b);
+        }
+
+        /* For A!=ACC or B!=ACC */
+        template <typename A, typename B, typename Accumulator>
+        inline Accumulator multiply(
+            A a, B b, typename std::enable_if<need_transform<A, B, Accumulator>>::type* = nullptr)
+        {
+            return static_cast<Accumulator>(a) * static_cast<Accumulator>(b);
+        }
+
```

Comment:
From a big-picture standpoint, I don't feel these are a good fit here.
* The new need_transform variable is now confusing because it has nothing to do with the above class Transform<T>.
* The new multiply functions are specific to the SolveCPU function and have no other use in the Client namespace
* The calling of a multiply function feels unnatural, as the function template a) cannot be automatically deduced and b) is not in operator form as would normally be expected.